annnufan/assignment1/solution_1.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <algorithm>
+#include <cctype>
+#include <functional>
+
+void word_anagram(bool case_sens) {
+	std::string word_1, word_2;
+	std::cin >> word_1 >> word_2;
+	if (!case_sens) {
+		std::transform(word_1.begin(), word_1.end(), word_1.begin(), std::ptr_fun <int, int> (std::tolower));
+		std::transform(word_2.begin(), word_2.end(), word_2.begin(), std::ptr_fun <int, int> (std::tolower));
+	}
+	if (word_1.length() != word_2.length()) {
+		std::cout << "Second word isn't anagram of first\n";
+		return;
+	}
+
+	std::vector<int> count(256, 0);
+	for (auto ch : word_1) {
+		count[ch]++;
+	}
+	for (auto ch : word_2) {
+		if (count[ch] > 0) {
+			count[ch]--;
+		} else {
+			std::cout << "Second word isn't anagram of first\n";

```

Comment:
Line 29 is a duplication of line 17.
This can be avoided when separating the in/out and the logic.